### PR TITLE
DOC: Fixing EX01 - Added examples

### PR DESCRIPTION
**Pull request by DeaMariaLeon**
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

Towards https://github.com/pandas-dev/pandas/issues/37875

### CODE REVIEW

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -6295,6 +6328,22 @@ def map(self, mapper, na_action: Literal["ignore"] | None = None):
...
+        --------
+        >>> idx = pd.Index([1, 2, 3])
+        >>> idx.map({1: 'a', 2: 'b', 3: 'c'})
+        Index(['a', 'b', 'c'], dtype='object')
+
+        Using `map` with a function:
+
+        >>> idx = pd.Index([1, 2, 3])
+        >>> idx.map('I am a {}'.format)
+        Index(['I am a 1', 'I am a 2', 'I am a 3'], dtype='object')
```
Comment (MarcoGorelli) [inline pandas/core/indexes/base.py:6342]: 😆 nice example

File: pandas/core/indexes/base.py [hunk 2]
```diff
@@ -2939,6 +2947,12 @@ def fillna(self, value=None, downcast=None):
         --------
         DataFrame.fillna : Fill NaN values of a DataFrame.
         Series.fillna : Fill NaN Values of a Series.
+
+        Examples
+        --------
+        >>> idx = pd.Index([None, np.nan, 3])
```
Comment (MarcoGorelli) [inline pandas/core/indexes/base.py:2953]: although you're constructing with `None` and `np.nan`, they both get converted to `np.nan`
```
In [2]:  pd.Index([None, np.nan, 3])
Out[2]: Index([nan, nan, 3.0], dtype='float64')
```

To reduce the chance for confusion, shall we just construct with `np.nan` directly?

File: pandas/core/indexes/base.py [hunk 3]
```diff
@@ -2969,6 +2983,12 @@ def dropna(self, how: AnyAll = "any") -> Self:
         Returns
         -------
         Index
+
+        Examples
+        --------
+        >>> idx = pd.Index([1, None, 3])
```
Comment (MarcoGorelli) [inline pandas/core/indexes/base.py:2989]: same here

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
nice!

File: pandas/core/indexes/base.py (hunk 2 above)
Comment (DeaMariaLeon) [reply pandas/core/indexes/base.py]: Sure

File: pandas/core/indexes/base.py (hunk 3 above)
Comment (DeaMariaLeon) [reply pandas/core/indexes/base.py]: Yes - thanks

File: pandas/core/indexes/base.py (hunk 1 above)
Comment (DeaMariaLeon) [reply pandas/core/indexes/base.py:6342]: There was something similar already in the docs. I stole it.  :-)

**Review by MarcoGorelli — APPROVED** [review]
thanks @DeaMariaLeon !

**Pull request merged**